File: app/src/main/java/com/sjn/stamp/db/Artist.java
package com.sjn.stamp.db;

import io.realm.annotations.Index;
import io.realm.annotations.PrimaryKey;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

@Accessors(prefix = "m")
@AllArgsConstructor(suppressConstructorProperties = true)
@NoArgsConstructor
@Getter
@Setter
public class Artist {
    @Index
    String mName;
    String mAlbumArtUri;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Artist)) return false;

        Artist artist = (Artist) o;

        return mName != null ? mName.equals(artist.mName) : artist.mName == null;

    }

    @Override
    public int hashCode() {
        return mName != null ? mName.hashCode() : 0;
    }
}

File: app/src/main/java/com/sjn/stamp/db/Artist.kt
package com.sjn.stamp.db

import io.realm.RealmObject
import io.realm.annotations.Index
import io.realm.annotations.PrimaryKey

open class Artist(
        @PrimaryKey var id: Long = 0,
        @Index var name: String? = null,
        var albumArtUri: String? = null
) : RealmObject() {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Artist) return false

        val artist = other as Artist?

        return if (name != null) name == artist!!.name else artist!!.name == null

    }

    override fun hashCode(): Int = if (name != null) name!!.hashCode() else 0
}